tests/test_bitsparsemap.py

```diff
@@ -0,0 +1,431 @@
+        )
+
+        sparse_map[1_000_000] = True
+        sparse_map[100_000_000] = True
+
+        fname = os.path.join(self.test_dir, f"healsparse_bitmask_{nside_map}_giant.hsp")
+        sparse_map.write(fname, clobber=True)
+        sparse_map_in = HealSparseMap.read(fname)
+
+        # Confirm it's the correct type.
+        self.assertTrue(sparse_map_in.is_bit_mask_map)
+        self.assertEqual(sparse_map_in.dtype, np.bool_)
+        self.assertEqual(sparse_map_in.sentinel, False)
+        # Confirm that it was reshaped on write.
+        self.assertIn('RESHAPED', sparse_map_in.metadata)
+        self.assertTrue(sparse_map_in.metadata['RESHAPED'])
+
+        self.assertEqual(sparse_map_in.n_valid, 2)
+
+        # Test reading back partial coverage.
+        cov_pixels = sparse_map._cov_map.cov_pixels([1_000_000, 100_000_000])
+
+        sparse_map_in_partial = healsparse.HealSparseMap.read(
+            fname,
+            pixels=cov_pixels,
+        )
+        self.assertTrue(sparse_map_in_partial.is_bit_mask_map)
+        self.assertEqual(sparse_map_in_partial.dtype, np.bool_)
+        self.assertEqual(sparse_map_in_partial.sentinel, False)
+
+        cov_pixels_partial, = np.where(sparse_map_in_partial.coverage_mask)
+        testing.assert_array_equal(cov_pixels_partial, cov_pixels)
+        testing.assert_array_equal(sparse_map_in_partial.valid_pixels, [1_000_000, 100_000_000])
+
+    @pytest.mark.skipif(not healsparse.parquet_shim.use_pyarrow, reason='Requires pyarrow')
+    def test_bit_mask_map_parquet_io(self):
+        pass
+
+    def test_bit_mask_map_fits_io_compression(self):
+        nside_coverage = 32
+        nside_map = 1024
+
+        self.test_dir = tempfile.mkdtemp(dir='./', prefix='TestHealSparse-')
+
+        sparse_map = HealSparseMap.make_empty(nside_coverage, nside_map, np.bool_, bit_mask=True)
+
+        sparse_map[10_000_000: 11_000_000] = True
+
+        fname = os.path.join(self.test_dir, f"healsparse_bitmask_{nside_map}.hsp")
+        fname_nocomp = os.path.join(
+            self.test_dir,
+            f"healsparse_bitmask_{nside_map}_nocomp.hsp",
+        )
+
+        sparse_map.write(fname, nocompress=False)
+        sparse_map.write(fname_nocomp, nocompress=True)
+
+        sparse_map_in = HealSparseMap.read(fname)
+        sparse_map_in_nocomp = HealSparseMap.read(fname_nocomp)
+
+        self.assertTrue(sparse_map_in_nocomp.is_bit_mask_map)
+        self.assertEqual(sparse_map_in_nocomp.dtype, np.bool_)
+        self.assertEqual(sparse_map_in_nocomp.sentinel, False)
+
+        testing.assert_array_equal(sparse_map_in_nocomp.valid_pixels, sparse_map_in.valid_pixels)
+
+    def test_bit_mask_fracdet(self):
+        nside_coverage = 32
+        nside_map = 2**13
+
+        # We compare the fracdet map generated with the bit_mask code to that
+        # generated with a regular boolean map.
+        sparse_map = HealSparseMap.make_empty(nside_coverage, nside_map, np.bool_, bit_mask=True)
```

Comment:
```suggestion
        sparse_map = HealSparseMap.make_empty(nside_coverage, nside_map, np.bool_, bit_packed=True)
```

This might be a slightly better name for the API option.